MapboxCoreNavigation/RouteProgress.swift

```diff
@@ -388,21 +388,50 @@ open class RouteStepProgress: NSObject {
     /**
      The next intersection the user will travel through.
      
-     The step must contains `Intersections` for this value not be `nil`.
+     The step must contain `Intersections` for this value not be `nil`.
      */
     @objc public var upcomingIntersection: Intersection? {
-        guard let intersections = intersectionsIncludingUpcomingManeuverIntersection, intersectionIndex + 1 < intersections.endIndex else {
+        guard let intersections = intersectionsIncludingUpcomingManeuverIntersection, intersections.startIndex..<intersections.endIndex-1 ~= intersectionIndex else {
```

Comment:
@1ec5 what do you mean by use the `...` operator? can you clarify please?

Reply:
These two lines have the same effect of returning 2, but the first is more straightforward:

```swift
(1...2).upperBound
(1..<3-1).upperBound
```

Reply:
@vincethecoder I think this is the last remaining issue to fix before merging.

Reply:
@1ec5 In this context, since we're doing:

Okay so using the `CountableClosedRange` you suggest, we would end up with this approach:
```
intersections.startIndex...intersections.endIndex-2 ~= intersectionIndex
```
This is a bit buggy because, we could end up with a single intersections array, which would end up with an invalid range where `upperBound` value could be lower than `lowerBound` value.

For example:
intersections with `count` 1 will end up like this: `(0...1-2) -> (0...-1)`. It's safer to use a `(0..<1-1) -> (0...0)` than the former.

cc @bsudekum

Reply:
Oops, you’re right, I was subtracting in the wrong direction. 😅